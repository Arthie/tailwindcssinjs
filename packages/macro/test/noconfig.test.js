const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");

pluginTester.default({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: {
    "no usage": `import tw from '../lib/macro'`,
    "correct usage 1": `
      import tw from '../lib/macro';
      const css = tw('w-1/2 bg-blue-500 scale-0 active:bg-red-300 lg:hover[text-red-200 bg-blue-500]', ['lg:text-red-300',"focus:bg-green-300"], "md:bg-green-300");
    `,
    "correct usage 2": `
      import tw from '../lib/macro';
      const css = tw("m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]");
    `,
    "correct usage tagged": `
      import tw from '../lib/macro';
      const css = tw\`m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]\`;
    `,
    components: `
      import tw from '../lib/macro';
      const css = tw\`container\`
    `,
    "variant class not allowed": {
      code: `
      import tw from '../lib/macro';
      const css = tw\`md[container]\`
      `,
      snapshot: true,
      error: /Variant class/,
    },
    "class not found error": {
      code: `
      import tw from '../lib/macro';
      const css = tw\`lol\`
      `,
      snapshot: true,
      error: /not found/,
    },
    "version 1.3.X test 1": `
      import tw from '../lib/macro';
      const css = tw\`group-focus:text-red-100 delay-1000 space-y-4 flex space-x-2 divide-x divide-y divide-blue-500 space-x-reverse\`
      `,
    "version 1.3.X test 2": `
      import tw from '../lib/macro';
      const css = tw\`group-focus:text-red-100 md:delay-1000 md:space-y-4 md:space-x-2 md:divide-x md:divide-y md:divide-blue-500 md:space-x-reverse\`
      `,
  },
});
