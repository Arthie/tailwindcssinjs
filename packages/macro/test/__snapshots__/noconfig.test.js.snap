// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros class not found error: class not found error 1`] = `

import tw from '../lib/macro';
const css = tw\`lol\`

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /home/arthie/projects/tailwindcssinjs/packages/macro/test/noconfig.test.js: Error: @tailwindcssinjs/macro - Class "lol" not found. Check Tailwind config. Restart dev server and or clearing babel cache after Tailwind config changes.

`;

exports[`macros components: components 1`] = `

import tw from '../lib/macro';
const css = tw\`container\`

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  width: "100%",
  "@media (min-width: 668px)": {
    maxWidth: "668px",
  },
  "@media (min-width: 868px)": {
    maxWidth: "868px",
  },
  "@media (min-width: 640px)": {
    maxWidth: "640px",
  },
  "@media (min-width: 1024px)": {
    maxWidth: "1024px",
  },
  "@media (min-width: 1280px)": {
    maxWidth: "1280px",
  },
  "@media (min-width: 1536px)": {
    maxWidth: "1536px",
  },
};


`;

exports[`macros correct usage 1: correct usage 1 1`] = `

import tw from '../lib/macro';
const css = tw('w-1/2 bg-blue-500 scale-0 active:bg-red-300 lg:hover[text-red-200 bg-blue-500]', ['lg:text-red-300',"focus:bg-green-300"], "md:bg-green-300");

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--bg-opacity": "1",
  "--transform-scale-x": "0",
  "--transform-scale-y": "0",
  width: "50%",
  backgroundColor: ["#3f83f8", "rgba(63, 131, 248, var(--bg-opacity))"],
  "&:active": {
    "--bg-opacity": "1",
    backgroundColor: ["#f8b4b4", "rgba(248, 180, 180, var(--bg-opacity))"],
  },
  "&:focus": {
    "--bg-opacity": "1",
    backgroundColor: ["#84e1bc", "rgba(132, 225, 188, var(--bg-opacity))"],
  },
  "@media (min-width: 1024px)": {
    "--bg-opacity": "1",
    backgroundColor: ["#84e1bc", "rgba(132, 225, 188, var(--bg-opacity))"],
  },
  "@media (min-width: 1280px)": {
    "--text-opacity": "1",
    "&:hover": {
      "--text-opacity": "1",
      color: ["#fbd5d5", "rgba(251, 213, 213, var(--text-opacity))"],
      "--bg-opacity": "1",
      backgroundColor: ["#3f83f8", "rgba(63, 131, 248, var(--bg-opacity))"],
    },
    color: ["#f8b4b4", "rgba(248, 180, 180, var(--text-opacity))"],
  },
};


`;

exports[`macros correct usage 2: correct usage 2 1`] = `

import tw from '../lib/macro';
const css = tw("m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]");

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--text-opacity": "1",
  "--transform-translate-x": "0",
  "--transform-translate-y": "-50%",
  "--transform-rotate": "0",
  "--transform-skew-x": "0",
  "--transform-skew-y": "0",
  "--transform-scale-x": "1",
  "--transform-scale-y": "1",
  margin: "0",
  marginRight: "1rem",
  transform:
    "translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y))",
  color: ["#f98080", "rgba(249, 128, 128, var(--text-opacity))"],
  "@media (min-width: 1280px)": {
    "&:hover": {
      "--bg-opacity": "1",
      backgroundColor: ["#e1effe", "rgba(225, 239, 254, var(--bg-opacity))"],
      "--text-opacity": "1",
      color: ["#fbd5d5", "rgba(251, 213, 213, var(--text-opacity))"],
    },
  },
};


`;

exports[`macros correct usage tagged: correct usage tagged 1`] = `

import tw from '../lib/macro';
const css = tw\`m-0 mr-4 text-red-400 transform -translate-y-1/2 lg:hover[bg-blue-100 text-red-200]\`;

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  "--text-opacity": "1",
  "--transform-translate-x": "0",
  "--transform-translate-y": "-50%",
  "--transform-rotate": "0",
  "--transform-skew-x": "0",
  "--transform-skew-y": "0",
  "--transform-scale-x": "1",
  "--transform-scale-y": "1",
  margin: "0",
  marginRight: "1rem",
  transform:
    "translateX(var(--transform-translate-x)) translateY(var(--transform-translate-y)) rotate(var(--transform-rotate)) skewX(var(--transform-skew-x)) skewY(var(--transform-skew-y)) scaleX(var(--transform-scale-x)) scaleY(var(--transform-scale-y))",
  color: ["#f98080", "rgba(249, 128, 128, var(--text-opacity))"],
  "@media (min-width: 1280px)": {
    "&:hover": {
      "--bg-opacity": "1",
      backgroundColor: ["#e1effe", "rgba(225, 239, 254, var(--bg-opacity))"],
      "--text-opacity": "1",
      color: ["#fbd5d5", "rgba(251, 213, 213, var(--text-opacity))"],
    },
  },
};


`;

exports[`macros no usage: no usage 1`] = `

import tw from '../lib/macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros variant class not allowed: variant class not allowed 1`] = `

import tw from '../lib/macro';
const css = tw\`md[container]\`

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /home/arthie/projects/tailwindcssinjs/packages/macro/test/noconfig.test.js: Error: @tailwindcssinjs/macro - Variant class "md" not allowed with class "container" of type "component"

`;

exports[`macros version 1.3.X test 1: version 1.3.X test 1 1`] = `

import tw from '../lib/macro';
const css = tw\`group-focus:text-red-100 delay-1000 space-y-4 flex space-x-2 divide-x divide-y divide-blue-500 space-x-reverse\`

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  transitionDelay: "1000ms",
  display: "flex",
  ".group:focus &": {
    "--text-opacity": "1",
    color: ["#fde8e8", "rgba(253, 232, 232, var(--text-opacity))"],
  },
  "& > :not(template) ~ :not(template)": {
    "--space-y-reverse": "0",
    "--space-x-reverse": "1",
    "--divide-x-reverse": "0",
    "--divide-y-reverse": "0",
    "--divide-opacity": "1",
    marginTop: "calc(1rem * calc(1 - var(--space-y-reverse)))",
    marginBottom: "calc(1rem * var(--space-y-reverse))",
    marginRight: "calc(0.5rem * var(--space-x-reverse))",
    marginLeft: "calc(0.5rem * calc(1 - var(--space-x-reverse)))",
    borderRightWidth: "calc(1px * var(--divide-x-reverse))",
    borderLeftWidth: "calc(1px * calc(1 - var(--divide-x-reverse)))",
    borderTopWidth: "calc(1px * calc(1 - var(--divide-y-reverse)))",
    borderBottomWidth: "calc(1px * var(--divide-y-reverse))",
    borderColor: ["#3f83f8", "rgba(63, 131, 248, var(--divide-opacity))"],
  },
};


`;

exports[`macros version 1.3.X test 2: version 1.3.X test 2 1`] = `

import tw from '../lib/macro';
const css = tw\`group-focus:text-red-100 md:delay-1000 md:space-y-4 md:space-x-2 md:divide-x md:divide-y md:divide-blue-500 md:space-x-reverse\`

      ↓ ↓ ↓ ↓ ↓ ↓

const css = {
  ".group:focus &": {
    "--text-opacity": "1",
    color: ["#fde8e8", "rgba(253, 232, 232, var(--text-opacity))"],
  },
  "@media (min-width: 1024px)": {
    transitionDelay: "1000ms",
    "& > :not(template) ~ :not(template)": {
      "--space-y-reverse": "0",
      marginTop: "calc(1rem * calc(1 - var(--space-y-reverse)))",
      marginBottom: "calc(1rem * var(--space-y-reverse))",
      "--space-x-reverse": "1",
      marginRight: "calc(0.5rem * var(--space-x-reverse))",
      marginLeft: "calc(0.5rem * calc(1 - var(--space-x-reverse)))",
      "--divide-x-reverse": "0",
      borderRightWidth: "calc(1px * var(--divide-x-reverse))",
      borderLeftWidth: "calc(1px * calc(1 - var(--divide-x-reverse)))",
      "--divide-y-reverse": "0",
      borderTopWidth: "calc(1px * calc(1 - var(--divide-y-reverse)))",
      borderBottomWidth: "calc(1px * var(--divide-y-reverse))",
      "--divide-opacity": "1",
      borderColor: ["#3f83f8", "rgba(63, 131, 248, var(--divide-opacity))"],
    },
  },
};


`;
